### PR TITLE
Vb 2467 add session template visit stats for admin

### PR DESCRIPTION
**Pull request by AledWEvans**
Add session template visit stats for admin

### CODE REVIEW

File: src/main/kotlin/uk/gov/justice/digital/hmpps/visitscheduler/repository/SessionTemplateRepository.kt [hunk 1]
```diff
@@ -16,6 +17,33 @@ import java.time.LocalTime
...
+  fun findValidSessionTemplatesBy(
+    @Param("reference") reference: String,
+    @Param("visitsFromDate") visitsFromDate: LocalDate,
+    @Param("visitsToDate") visitsToDate: LocalDate,
+  ): Tuple
+
+  @Query(
+    "select count(*) from visit v " +
+      " JOIN session_template st ON st.reference = v.session_template_reference " +
+      " WHERE st.reference = :reference AND v.visit_start > :visitsFromDate AND v.visit_start < :visitsToDate ",
```
Comment (dhirajshettymoj) [inline src/main/kotlin/uk/gov/justice/digital/hmpps/visitscheduler/repository/SessionTemplateRepository.kt:38]: visit_status as BOOKED or BOOKED and RESERVED?

File: src/main/kotlin/uk/gov/justice/digital/hmpps/visitscheduler/dto/sessions/RequestSessionTemplateVisitStatsDto.kt [hunk 2]
```diff
@@ -0,0 +1,11 @@
+package uk.gov.justice.digital.hmpps.visitscheduler.dto.sessions
+
+import io.swagger.v3.oas.annotations.media.Schema
+import jakarta.validation.constraints.NotNull
+import java.time.LocalDate
+
+data class RequestSessionTemplateVisitStatsDto(
+  @Schema(description = "Visits from date stats", example = "2019-11-02", required = true)
+  @field:NotNull
+  val visitsFromDate: LocalDate,
```
Comment (dhirajshettymoj) [inline src/main/kotlin/uk/gov/justice/digital/hmpps/visitscheduler/dto/sessions/RequestSessionTemplateVisitStatsDto.kt:10]: what if we allow a visitsToDate - nullable as well but default it to n days?

File: src/main/kotlin/uk/gov/justice/digital/hmpps/visitscheduler/repository/SessionTemplateRepository.kt [hunk 3]
```diff
@@ -16,6 +17,33 @@ import java.time.LocalTime
...
+  @Query(
+    "SELECT MAX(open),MAX(closed) FROM" +
+      "(SELECT  COUNT(CASE WHEN v.visit_restriction = 'OPEN' THEN 1 END) AS open, " +
+      " COUNT(CASE WHEN v.visit_restriction = 'CLOSED' THEN 1 END) AS closed  FROM visit v " +
+      " JOIN session_template st ON st.reference = v.session_template_reference " +
+      " WHERE st.reference = :reference AND v.visit_start > :visitsFromDate AND v.visit_start < :visitsToDate " +
+      " GROUP BY v.visit_start ) AS tmp ",
+    nativeQuery = true,
+  )
+  fun findValidSessionTemplatesBy(
```
Comment (dhirajshettymoj) [inline src/main/kotlin/uk/gov/justice/digital/hmpps/visitscheduler/repository/SessionTemplateRepository.kt:29]: could the method name have capacity?

Comment (AledWEvans) [reply src/main/kotlin/uk/gov/justice/digital/hmpps/visitscheduler/repository/SessionTemplateRepository.kt:29]: yeha bad naming

File: src/main/kotlin/uk/gov/justice/digital/hmpps/visitscheduler/dto/sessions/RequestSessionTemplateVisitStatsDto.kt (hunk 2 above)
Comment (AledWEvans) [reply src/main/kotlin/uk/gov/justice/digital/hmpps/visitscheduler/dto/sessions/RequestSessionTemplateVisitStatsDto.kt:10]: not sure if we need that,...

**Review by dhirajshettymoj — APPROVED** [review]

**Pull request merged**